### PR TITLE
Add authority_map feature

### PR DESCRIPTION
**Pull request by nikosev**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,12 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+### Added
+
+- Replace modified entityIDs with the old ones
```
Comment (NicolasLiampotis) [inline CHANGELOG.md]: Add support for migrating users from IdPs with updated entityID

File: lib/Auth/Process/OpaqueSmartID.php [hunk 2]
```diff
@@ -331,6 +344,13 @@ private function generateUserId($attributes, $request)
             if ($this->addAuthority) {
                 $authority = $this->getAuthority($request);
             }
+            if (!empty($authority) && array_key_exists($authority, $this->authorityMap)) {
+                Logger::debug(
```
Comment (NicolasLiampotis) [inline lib/Auth/Process/OpaqueSmartID.php]: change to info

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (nikosev) [reply CHANGELOG.md]: resolved by commit [Update CHANGELOG description for authority_map feature](https://github.com/rciam/simplesamlphp-module-userid/pull/14/commits/115c988fc90ef93fde2f7f2e7e8a37b36e2c2980)

File: lib/Auth/Process/OpaqueSmartID.php (hunk 2 above)
Comment (nikosev) [reply lib/Auth/Process/OpaqueSmartID.php]: resolved by commit [Change log lever when the module is using an entityID from authorityMap](https://github.com/rciam/simplesamlphp-module-userid/pull/14/commits/7166c21ee53c68266ef8eb9519bd44cb6b9274fa)

**Review by NicolasLiampotis — APPROVED** [review]
LGTM

**Pull request merged**